include/envoy/server/configuration.h

```diff
@@ -90,9 +93,27 @@ class Main {
   virtual std::chrono::milliseconds statsFlushInterval() const PURE;
 
   /**
-   * @return const Watchdog& the configuration of the watchdog.
+   * @return WatchdogOptConstRef the configuration of the watchdog.
+   *         If multiWatchdog() returns true this will be empty.
+   */
+  virtual WatchdogOptConstRef watchdogConfig() const PURE;
+
+  /**
+   * @return WatchdogOptConstRef the configuration of the auxiliary watchdog.
+   *         This will be empty unless we're multiWatchdog() returns true.
+   */
+  virtual WatchdogOptConstRef auxWatchdogConfig() const PURE;
+
+  /**
+   * @return WatchdogOptConstRef the configuration of the worker watchdog.
+   *         This will be empty unless we're multiWatchdog() returns true.
+   */
+  virtual WatchdogOptConstRef workerWatchdogConfig() const PURE;
+
+  /**
+   * @return Whether we're using multiWatchdog.
    */
-  virtual const Watchdog& watchdogConfig() const PURE;
+  virtual bool multiWatchdog() const PURE;
```

Comment:
Requiring callers to interpret the results of calling a method by consulting a different method seems like an API smell. Would it be possible to get rid of `watchdogConfig()` and require callers to always treat `auxWatchdogConfig()` and `workerWatchdogConfig()` as distinct, even if they actually return the same config?

Reply:
The problem is that there's no way to perfectly emulate the single watchdog case with multiple watchdogs, right?

Reply:
Yep. Shouldn't be an issue now though with always having multiple guarddogs in the system.